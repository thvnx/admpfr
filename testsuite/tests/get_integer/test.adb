with Ada.Text_IO;    use Ada.Text_IO;
with Admpfr;         use Admpfr;

procedure Test is
   M   : Mpfloat;
   I   : Long_Integer;
begin
   M.Set (2);
   I := M.Get_Long_Integer;
   Put_Line (I'Image);
   Put_Line (M.Fits_Long_Integer'Image);

   M.Set (Float (0.1));
   I := M.Get_Long_Integer;
   Put_Line (I'Image);
   Put_Line (M.Fits_Long_Integer'Image);
   --  TODO: test inexact flag

   M.Set_Nan;
   I := M.Get_Long_Integer;
   Put_Line (I'Image);
   Put_Line (M.Fits_Long_Integer'Image);
   --  TODO: test erange flag

   M.Set ("1e999999");
   I := M.Get_Long_Integer;
   Put_Line (I'Image);
   Put_Line (Long_Integer'Last'Image);
   Put_Line (M.Fits_Integer'Image);
   Put_Line (M.Fits_Long_Integer'Image);
   --  TODO: test erange flag

   M.Set (Long_Integer'Last, RNDD);
   I := M.Get_Long_Integer;
   Put_Line (I'Image);
   Put_Line (M.Fits_Integer'Image);
   Put_Line (M.Fits_Long_Integer'Image);
end Test;
