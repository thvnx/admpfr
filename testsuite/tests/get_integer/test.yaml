driver: admpfr
control:
- [XFAIL, "env.build.os.name == 'windows'",
   "windows long is 4 bytes, output expects 8 bytes long values"]
