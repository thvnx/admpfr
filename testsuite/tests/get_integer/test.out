 2
 0
 0
 9223372036854775807
 9223372036854775807
